From 7a8cce1266ae9c8ffff153b936b9647e23e84132 Mon Sep 17 00:00:00 2001
From: IOhannes m zmoelnig <zmoelnig@umlautQ.umlaeute.mur.at>
Date: Thu, 6 Jul 2017 23:32:10 +0200
Subject: [PATCH 1/2] check for NULL-functionpointers into jack-framework on
 all platforms

sometimes, jack is also weakly linked on linux (and protecting the
check with __APPLE__ leads to crashes in such situations)

in any case, we want to be sure that jack_client_open is non-NULL, in *all* cases...
---
 src/s_audio_jack.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/s_audio_jack.c b/src/s_audio_jack.c
index 217692ea..028fadcc 100644
--- a/src/s_audio_jack.c
+++ b/src/s_audio_jack.c
@@ -307,13 +307,11 @@ jack_open_audio(int inchans, int outchans, int rate, t_audiocallback callback)
     int srate;
     jack_status_t status;

-#ifdef __APPLE__
     if (!jack_client_open)
     {
-        error("Can't open Jack (it seems not to be installed on this Mac)");
+        error("Can't open Jack (it seems not to be installed)");
         return 1;
     }
-#endif

     jack_dio_error = 0;


From 2fa1b8e84982a212b7e7a0ded47bc90de4edcab9 Mon Sep 17 00:00:00 2001
From: danomatika <danomatika@gmail.com>
Date: Mon, 10 Jul 2017 03:22:09 +0200
Subject: [PATCH 2/2] only include jack/weakjack.h on macOS, fixes segfault
 when using jack on Linux

---
 src/s_audio_jack.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/s_audio_jack.c b/src/s_audio_jack.c
index 028fadcc..2907e87d 100644
--- a/src/s_audio_jack.c
+++ b/src/s_audio_jack.c
@@ -12,7 +12,9 @@
 #include <string.h>
 #include "m_pd.h"
 #include "s_stuff.h"
+#ifdef __APPLE__
 #include <jack/weakjack.h>
+#endif
 #include <jack/jack.h>
 #include <regex.h>
 
